### PR TITLE
docs:PE-2831 provision static IP

### PR DESCRIPTION
**Pull request by lennessyy**
Document the Static IP provision changes for Edge Native cluster setup

## Describe the Change

This PR adds the steps to provision a static IP address for Edge hosts when a cluster is first created and when someone adds an Edge host to an existing cluster. 

## Review Changes

💻 [Preview - Cluster Deployment](https://deploy-preview-1726--docs-spectrocloud.netlify.app/clusters/edge/site-deployment/site-installation/cluster-deployment)

🎫 [PE-2831](https://spectrocloud.atlassian.net/browse/PE-2831)

### CODE REVIEW

**Review by vipsharm — APPROVED** [review]

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md [hunk 1]
```diff
@@ -14,8 +14,13 @@ To complete the Edge Installation process, an Edge host must become a member of
 Select the workflow that best fits your needs.
 
 - [Create an Edge Native Host Cluster](#create-an-edge-native-host-cluster)
-
+  - [Prerequisites](#prerequisites)
+  - [Create Cluster](#create-cluster)
+  - [Validate](#validate)
 - [Add an Edge Host to a Host Cluster](#add-an-edge-host-to-a-host-cluster)
+  - [Prerequisites](#prerequisites-1)
+  - [Add Edge Host to Node Pool](#add-edge-host-to-node-pool)
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:22]:
```suggestion
  - [Add an Edge Host to a Node Pool](#add-edge-host-to-node-pool)
```

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md [hunk 2]
```diff
@@ -14,8 +14,13 @@ To complete the Edge Installation process, an Edge host must become a member of
...
+  - [Validate](#validate-1)
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:23]: We have a TOC on the right-hand side nav menu. We don't need this, nor is this page large enough to warrant one.

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md [hunk 3]
```diff
@@ -55,11 +60,20 @@ Use the following steps to create a new host cluster so that you can add Edge ho
 
 9. The node configuration page is where you can specify what Edge hosts make up the host cluster. Assign Edge hosts to the **master-pool** and the **worker-pool**. When you have completed configuring the node pools, click on **Next**.
 
+10. (Optional) When you assign Edge hosts to node pools, you can optionally specify a static IP address for each Edge host. If you want to specify a static IP, toggle on **Static IP** and provide the following information:
+    
+    -  IP address for the Edge host
+    -  IP address for the default gateway
+    -  Subnet mask of your network
+    -  IP address of your DNS server
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:68]: We usually use a table with the columns "Field", and "Description" to explain fields. Use the description to provide more context.

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md [hunk 4]
```diff
@@ -119,8 +133,16 @@ To learn more, check out the resource from the etcd documentation titled [Why an
...
 7. Navigate to the **Edge Hosts drop-down Menu** and select your Edge host.
 
+8. (Optional) When you select your Edge host, you can optionally specify a static IP address for the Edge host. If you want to specify a static IP, toggle on **Static IP** and provide the following information:
+    
+    -  IP address for the Edge host
+    -  IP address for the default gateway
+    -  Subnet mask of your network
+    -  IP address of your DNS server
+
+    If certain network information is already available, the corresponding fields will be pre-populated. After you add the Edge host to your cluster, you will not be able to change its IP address unless you remove and re-add them back to the cluster.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:143]: Same feedback as above

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md [hunk 5]
```diff
@@ -55,11 +60,20 @@ Use the following steps to create a new host cluster so that you can add Edge ho
...
 9. The node configuration page is where you can specify what Edge hosts make up the host cluster. Assign Edge hosts to the **master-pool** and the **worker-pool**. When you have completed configuring the node pools, click on **Next**.
 
+10. (Optional) When you assign Edge hosts to node pools, you can optionally specify a static IP address for each Edge host. If you want to specify a static IP, toggle on **Static IP** and provide the following information:
+    
+    -  IP address for the Edge host
+    -  IP address for the default gateway
+    -  Subnet mask of your network
+    -  IP address of your DNS server
+
+    If certain network information is already available, the corresponding fields will be pre-populated. After you create the cluster, you will not be able to change the IP address of your existing Edge hosts unless you remove and re-add them back to the cluster.
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:70]: The later part, should we add a caution box?

**Review by caroldelwing — APPROVED** [review]
A small suggestion.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Changes look good overall, just some action items that need to be addressed.

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md (hunk 2 above)
Comment (lennessyy) [reply docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:23]: I am not really sure how this happened - I never added these myself but they just appeared so I thought it was some Docusarus thing that we did. Now I wonder if it came from the Markdown extension that I installed.  I'll delete this.

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md [hunk 6]
```diff
@@ -55,11 +51,26 @@ Use the following steps to create a new host cluster so that you can add Edge ho
 
 9. The node configuration page is where you can specify what Edge hosts make up the host cluster. Assign Edge hosts to the **master-pool** and the **worker-pool**. When you have completed configuring the node pools, click on **Next**.
 
+10. (Optional) When you assign Edge hosts to node pools, you can optionally specify a static IP address for each Edge host. If you want to specify a static IP, toggle on **Static IP** and provide the following information:
 
-10. The Settings page is where you can configure a patching schedule, security scans, backup settings, and set up Role-Based Access Control (RBAC). Review the settings and make changes if needed. Click on **Validate**.
+  | **Field** | **Description** |
+  |-------|-------------|
+  | IP Address | The static IP address assigned to your Edge host. This should be unique within your network. |
+  | Default gateway | The IP address of the default gateway for your cluster network. This gateway routes traffic from your cluster to external networks, including the internet. |
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:59]:
```suggestion
  | Default gateway | The IP address of the default gateway for your cluster network. This gateway routes traffic from your cluster to external networks. |
```
Let's remove the internet statement. Internet access is an assumption that could cause concerns. In reality, this is controlled by the customer's network configuration.

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md [hunk 7]
```diff
@@ -119,8 +130,22 @@ To learn more, check out the resource from the etcd documentation titled [Why an
...
+  | Subnet mask | The subnet mask of your cluster network. This defines the range of IP addresses within your cluster network. |
+  | DNS server | The IP address of the DNS server your cluster uses for domain resolution. | 
+
+  If certain network information is already available, the corresponding fields will be pre-populated. 
+  
+:::caution
+  After you add the Edge host to your cluster, you will not be able to change its IP address unless you remove and re-add them back to the cluster. 
+:::
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:146]: Lenny, play with this to get it lined up with the bullet. You may have to tab the whole component, or create a `br /` and then add the element below the breakline followed with a tab.

CleanShot 2023-10-30 at 14 26 02

File: docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md [hunk 8]
```diff
@@ -119,8 +130,22 @@ To learn more, check out the resource from the etcd documentation titled [Why an
 
 7. Navigate to the **Edge Hosts drop-down Menu** and select your Edge host.
 
+8. (Optional) When you select your Edge host, you can optionally specify a static IP address for the Edge host. If you want to specify a static IP, toggle on **Static IP** and provide the following information:
+    
+  | **Field** | **Description** |
+  |-------|-------------|
+  | IP Address | The static IP address assigned to your Edge host. This should be unique within your network. |
+  | Default gateway | The IP address of the default gateway for your cluster network. This gateway routes traffic from your cluster to external networks, including the internet. |
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/cluster-deployment.md:138]:
```suggestion
  | Default gateway | The IP address of the default gateway for your cluster network. This gateway routes traffic from your cluster to external networks. |
```

**Review by karl-cardenas-coding — APPROVED** [review]
Looks great, just some last quality of life suggestions.

**Pull request merged**